riff-raff/app/deployment/Deployments.scala

```diff
@@ -91,27 +89,11 @@ class Deployments(deploymentEngine: DeploymentEngine,
     DeployRecord(new DateTime(), uuid, parameters, metaData.getOrElse(Map.empty[String, String]))
   }
 
-  lazy val completed: Observable[UUID] = deployCompleteSubject
-  private lazy val deployCompleteSubject = Subject[UUID]()
-
-  private val messagesSubscription: Subscription =
-    DeployReporter.messages.subscribe{ wrapper =>
-      try {
-        update(wrapper)
-      } catch {
-        case NonFatal(t) =>
-          log.error(s"Exception thrown whilst processing update with $wrapper", t)
-      }
-    }
-  def init() {}
-  def shutdown() { messagesSubscription.unsubscribe() }
-
-
   val deploysEnabled = new Switch(startAsOn = true)
```

Comment:
I noticed that this is using `akka.util.Switch`, which seems to provide pretty similar functionality to some of the code that is being ported across from Guardian Management - I wonder if we could simplify/make this consistent in the future

Reply:
Yeah, I think so. It was surprising to me that both were used. This PR is already quite large(!) but this is a good candidate for a follow-up.